File: wc107/case2/case2.go
package main

import (
	"fmt"
)

func minFlipsMonoIncr(S string) int {
	var cz, co int
	var i, j = 0, len(S) - 1
	for i < j {
		if S[i] == '0' {
			i++
		} else if S[j] == '1' {
			j--
		} else {
			break
		}
	}
	var ss = S[i : j+1]
	var left, right = 0, len(ss) - 1
	for left < right {
		if ss[left] == '1' {
			left++
		}
	}

}

func main() {
	fmt.Println(minFlipsMonoIncr("00110"))
	fmt.Println(minFlipsMonoIncr("010110"))
	fmt.Println(minFlipsMonoIncr("00011000"))
	fmt.Println(minFlipsMonoIncr("0101100011"))
	fmt.Println(minFlipsMonoIncr("10011111110010111011"))
	fmt.Println(minFlipsMonoIncr("11111"))
}

File: wc125/case2.cpp
#include <iostream>
#include <unordered_map>
#include <unordered_set>
#include <vector>

using namespace std;

class Solution {
public:
    int numRookCaptures(vector<vector<char>>& board) {
        // find rook
        int r, c;

        for (int i = 0; i < 8; ++i) {
            for (int j = 0; j < 8; ++j) {
                if (board[i][j] == 'R') {
                    r = i;
                    c = j;
                }
            }
        }
        int count = 0;
        // check 4 directions
        // check left
        for (int i = 1; c - i >= 0; ++i) {
            if (board[r][c - i] == 'p') {
                ++count;
                break;
            } else if (board[r][c - i] == 'B') {
                break;
            }
        }

        // check right
        for (int i = 1; c + i < 8; ++i) {
            if (board[r][c + i] == 'p') {
                ++count;
                break;
            } else if (board[r][c + i] == 'B') {
                break;
            }
        }

        // check top
        for (int i = 1; r + i < 8; ++i) {
            if (board[r + i][c] == 'p') {
                ++count;
                break;
            } else if (board[r + i][c] == 'B') {
                break;
            }
        }

        // check right
        for (int i = 1; r - i >= 0; ++i) {
            if (board[r - i][c] == 'p') {
                ++count;
                break;
            } else if (board[r - i][c] == 'B') {
                break;
            }
        }
        return count;
    }
};

int main() {
    Solution s;
    // s.numRookCaptures()
}